src/renderers/shared/fiber/ReactFiberReconciler.js

```diff
@@ -170,13 +174,13 @@ module.exports = function<T, P, I, TI, PI, C, CX>(config : HostConfig<T, P, I, T
 
     deferredUpdates,
 
-    getPublicRootInstance(container : OpaqueNode) : (ReactComponent<any, any, any> | I | TI | null) {
+    getPublicRootInstance(container : OpaqueNode) : (ReactComponent<any, any, any> | PI | null) {
       const root : FiberRoot = (container.stateNode : any);
       const containerFiber = root.current;
       if (!containerFiber.child) {
         return null;
       }
-      return containerFiber.child.stateNode;
+      return getPublicInstance(containerFiber.child.stateNode);
```

Comment:
This still suffers from this problem: https://github.com/facebook/react/pull/8628/files/9a9276b3fbcec4cfb6d5c69e891c5feafe0195f2#r95653360 Which remains a problem in commitWork.

You need to check whether `containerFiber.child.tag === HostComponent` before passing it to getPublicInstance.

Otherwise, in the case of `render(<Foo />)` you will pass an instance of `Foo` to `getPublicInstance` which is neither an `I` nor `TI` so the function doesn't know what to do with it.